### PR TITLE
Add test to expand_log with factor = True/False.

### PR DESCRIPTION
**Pull request by sbt4104**
#### References to other Issues or PRs
Fixes #18927 
#### Brief description of what is fixed or changed
Add test for expand_log with factor= True/False.

### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/functions/elementary/tests/test_exponential.py [hunk 1]
```diff
@@ -645,6 +645,18 @@ def test_issue_8866():
...
 
+def test_log_expand_factor():
+    assert (log(18)/log(3) - 2).expand(factor=True) == log(2)/log(3)
+    assert (log(12)/log(2)).expand(factor=True) == log(3)/log(2) + 2
+    assert (log(15)/log(3)).expand(factor=True) == 1 + log(5)/log(3)
+    assert (log(2)/(-log(12) + log(24))).expand(factor=True) == 1
+
+    assert expand_log(log(12), factor=True) == log(3) + 2*log(2)
+    assert expand_log(log(21)/log(7), factor=False) == (log(3) + log(7))/log(7)
+    assert expand_log(log(45)/log(5) + 4, factor=False) == (log(5) + 2*log(3))/log(5) + 4
```
Comment (smichr) [inline sympy/functions/elementary/tests/test_exponential.py]:
```suggestion
    assert expand_log(log(45)/log(5) + log(20) + log(4), factor=False) == (log(5) + 2*log(3))/log(5) + log(20) + 2*log(2)
```

File: sympy/functions/elementary/tests/test_exponential.py [hunk 2]
```diff
@@ -645,6 +645,18 @@ def test_issue_8866():
...
 
+def test_log_expand_factor():
+    assert (log(18)/log(3) - 2).expand(factor=True) == log(2)/log(3)
+    assert (log(12)/log(2)).expand(factor=True) == log(3)/log(2) + 2
+    assert (log(15)/log(3)).expand(factor=True) == 1 + log(5)/log(3)
+    assert (log(2)/(-log(12) + log(24))).expand(factor=True) == 1
+
+    assert expand_log(log(12), factor=True) == log(3) + 2*log(2)
+    assert expand_log(log(21)/log(7), factor=False) == (log(3) + log(7))/log(7)
+    assert expand_log(log(45)/log(5) + log(20) + log(4), factor=False) == (log(5) + 2*log(3))/log(5) + log(20) + 2*log(2)
```
Comment (Sc0rpi0n101) [inline sympy/functions/elementary/tests/test_exponential.py]: This line is longer than 79 characters.
Can you split it into two lines?

Check [PEP-8](https://www.python.org/dev/peps/pep-0008/#maximum-line-length) for reference.

File: sympy/functions/elementary/tests/test_exponential.py [hunk 3]
```diff
@@ -645,6 +645,18 @@ def test_issue_8866():
...
+    assert expand_log(log(45)/log(5) + log(26), factor=True) == log(2) + log(13) + (log(5) + 2*log(3))/log(5)
```
Comment (Sc0rpi0n101) [inline sympy/functions/elementary/tests/test_exponential.py]: This one too

File: sympy/functions/elementary/tests/test_exponential.py (hunk 2 above)
Comment (sbt4104) [reply sympy/functions/elementary/tests/test_exponential.py]: I have made the changes, please check.

**Comment by smichr** [comment]
I wonder if the following tweak makes sense: the `factor=False` is working to make a smarter change more often. If we only do so if things actually become simpler that might be better:
```python
>>> expand_log(log(15)/log(3)+log(20)/log(3))
1 + log(5)/log(3) + log(20)/log(3)
```
with
```diff
diff --git a/sympy/core/function.py b/sympy/core/function.py
index 482cfe1..7a2cbd4 100644
--- a/sympy/core/function.py
+++ b/sympy/core/function.py
@@ -2836,10 +2836,16 @@ def expand_log(expr, deep=True, force=False, factor=False):
     """
     from sympy import Mul, log
     if factor is False:
+        def _handle(x):
+            x1 = expand_log(x, deep=deep, force=force, factor=True)
+            x2 = expand_mul(x1)
+            if x2.count_ops() <= x1.count_ops():
+                    return x2
+            return x
         expr = expr.replace(
         lambda x: x.is_Mul and all(any(isinstance(i, log) and i.args[0].is_Rational
         for i in Mul.make_args(j)) for j in x.as_numer_denom()),
-        lambda x: expand_log(x, deep=deep, force=force, factor=True))
+        lambda x: _handle(x))

     return sympify(expr).expand(deep=deep, log=True, mul=False,
         power_exp=False, power_base=False, multinomial=False,
```

**Comment by sbt4104** [comment]
I think we should track log terms before and after the expand to only keep the simplified one.

```diff
--- a/sympy/core/function.py
+++ b/sympy/core/function.py
@@ -2836,10 +2836,16 @@ def expand_log(expr, deep=True, force=False, factor=False):
     """
     from sympy import Mul, log
     if factor is False:
+        def _handle(x):
+            x1 = expand_mul(expand_log(x, deep=deep, force=force, factor=True))
+            if x1.count(log) <= x.count(log):
+                return x1
+            return x
+
         expr = expr.replace(
         lambda x: x.is_Mul and all(any(isinstance(i, log) and i.args[0].is_Rational
         for i in Mul.make_args(j)) for j in x.as_numer_denom()),
-        lambda x: expand_log(x, factor=True))
+        lambda x: _handle(x))
 
     return sympify(expr).expand(deep=deep, log=True, mul=False,
         power_exp=False, power_base=False, multinomial=False,
```

**Pull request merged**